File: java/Mahasiswa.java
public class Mahasiswa { private String nama; private String nim;

    public Mahasiswa(String nama, String nim) {
        this.setNama(nama);
        this.setNim(nim);
    }

    public String getNama() {
        return nama;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public String getNim() {
        return nim;
    }

    public void setNim(String nim) {
        if (nim.matches("\\d{11}")) {
            this.nim = nim;
        } else {
            System.out.println("Data Mahasiswa");
        }
    }

    public static void main(String[] args) {
        Mahasiswa m = new Mahasiswa("Angeli Khairunissa", "32602400073");
        System.out.println("Nama : " + m.getNama());
        System.out.println("NIM  : " + m.getNim());
    }

}

File: python/mahasiswa.py
class Mahasiswa:
    def __init__(self, nama, nim):
        self.__nama = nama
        self.__nim = None
        self.nim = nim

    @property
    def nama(self):
        return self.__nama

    @nama.setter
    def nama(self, value):
        self.__nama = value

    @property
    def nim(self):
        return self.__nim

    @nim.setter
    def nim(self, value):
        if value.isdigit() and len(value) == 10:
            self.__nim = value
        else:
            print("Data Mahasiswa")

m = Mahasiswa("Angeli Khairunissa", "32602400073")
print("Nama :", m.nama)
print("NIM  :", m.nim)
